midp/sensor.js

```diff
@@ -0,0 +1,300 @@
+    model.klass.classInfo.getField("I.errorMsgs.[Ljava/lang/String;").set(model, J2ME.newStringArray(0));
+
+    var n = m.properties.length;
+    var p = J2ME.newStringArray(n);
+    for (var i = 0; i < n; i++) {
+        p[i] = J2ME.newString(m.properties[i]);
+    }
+    model.klass.classInfo.getField("I.properties.[Ljava/lang/String;").set(model, p);
+};
+
+Native["com/sun/javame/sensor/ChannelImpl.doGetChannelModel.(IILcom/sun/javame/sensor/ChannelModel;)V"] = function(sensorsNumber, number, model) {
+    if (sensorsNumber !== 0) {
+        console.error("Invalid sensor number: " + sensorsNumber);
+        return;
+    }
+    if (number < 0 || number >= AccelerometerSensor.channels.length) {
+        console.error("Invalid channel number: " + number);
+        return;
+    }
+    var c = AccelerometerSensor.channels[number];
+    model.klass.classInfo.getField("I.scale.I")
+               .set(model, c.scale);
+    model.klass.classInfo.getField("I.name.Ljava/lang/String;")
+               .set(model, J2ME.newString(c.name));
+    model.klass.classInfo.getField("I.unit.Ljava/lang/String;")
+               .set(model, J2ME.newString(c.unit));
+    model.klass.classInfo.getField("I.dataType.I")
+               .set(model, c.dataType);
+    model.klass.classInfo.getField("I.accuracy.I")
+               .set(model, c.accuracy);
+    model.klass.classInfo.getField("I.mrangeCount.I")
+               .set(model, c.mrangeArray.length);
+
+    var n = c.mrangeArray.length;
+    var array = J2ME.newArray(J2ME.PrimitiveArrayClassInfo.J.klass, n);
+    for (var i = 0; i < n; i++) {
+        array[i] = c.mrangeArray[i];
+    }
+    model.klass.classInfo.getField("I.mrageArray.[J").set(model, array);
+};
+
+Native["com/sun/javame/sensor/NativeSensor.doIsAvailable.(I)Z"] = function(number) {
+    // Only support the acceleration sensor with number = 0.
+    return number === 0 ? 1 : 0;
+};
+
+Native["com/sun/javame/sensor/NativeSensor.doInitSensor.(I)Z"] = function(number) {
+    if (number !== 0) {
+        return 0;
+    }
+    AccelerometerSensor.open();
+    return 1;
+};
+
+Native["com/sun/javame/sensor/NativeSensor.doFinishSensor.(I)Z"] = function(number) {
+    if (number !== 0) {
+        return 0;
+    }
+    AccelerometerSensor.close();
+    return 1;
+};
+
+Native["com/sun/javame/sensor/NativeChannel.doMeasureData.(II)[B"] = function(sensorNumber, channelNumber) {
+    if (sensorNumber !== 0 || channelNumber < 0 || channelNumber >= 3) {
+        if (sensorNumber !== 0) {
+            console.error("Invalid sensor number: " + sensorsNumber);
+        } else {
+            console.error("Invalid channel number: " + channelNumber);
+        }
+        return J2ME.newArray(J2ME.PrimitiveArrayClassInfo.B.klass, 0);
+    }
+
+    asyncImpl("[B", new Promise(function(resolve, reject) {
+        var result = AccelerometerSensor.readBuffer(channelNumber);
+        setTimeout(resolve.bind(null, result), 50);
```

Comment:
Maybe we should wait for a devicemotion event to occur before resolving this? (and, on desktop, use the timeout)